src/v/storage/log_manager.cc

```diff
@@ -401,6 +402,69 @@ ss::future<> log_manager::remove(model::ntp ntp) {
     });
 }
 
+/// Parse partition directory name
+static std::optional<std::pair<model::partition_id, model::revision_id>>
+parse_partition_directory(const ss::sstring& name) {
+    const std::regex re(R"(^(\d+)_(\d+)$)");
```

Comment:
nit: can we make this static thread_local so it doesn't have to be compiled every time ?